R/ggs_volcano.R

```diff
@@ -0,0 +1,181 @@
+#' data(res_de_macrophage, package = "GeneTonic")
+#' res_de <- res_macrophage_IFNg_vs_naive
+#'
+#' # res_enrich object
+#' data(res_enrich_macrophage, package = "GeneTonic")
+#' res_enrich <- shake_topGOtableResult(topgoDE_macrophage_IFNg_vs_naive)
+#' res_enrich <- get_aggrscores(res_enrich, res_de, anno_df)
+#'
+#' ggs_volcano(res_de,
+#'            res_enrich,
+#'            anno_df,
+#'            geneset_id = res_enrich$gs_id[1]
+#'            )
+#'            
+#' alternatively
+#'
+#' # genelist <- sapply(anno_df$gene_name, function(x) str_contains(x, "CD"))
+#' 
+#' # ggs_volcano(res_de,
+#' #             res_enrich,
+#' #             anno_df,
+#' #             genelist = genelist
+#' #             )
+#' 
+ggs_volcano <- function(res_de,
+                       res_enrich,
+                       annotation_obj = NULL,
+                       gtl = NULL,
+                       geneset_id = NULL,
+                       genelist = NULL,
+                       FDR = 0.05,
+                       color = "red",
+                       volcano_labels = 25,
+                       plot_title = NULL
+                       ) {
+
+  if (!is.null(gtl)) {
+    checkup_gtl(gtl)
+    dds <- gtl$dds
+    res_de <- gtl$res_de
+    res_enrich <- gtl$res_enrich
+    annotation_obj <- gtl$annotation_obj
+  }
+
+  # Retrieve information about genes in geneset gs_id
+  if (!is.null(geneset_id)) {
+    if (geneset_id %in% res_enrich[["gs_id"]]) {
+      thisset_name <- res_enrich[geneset_id, "gs_description"]
+      thisset_members <- unlist(strsplit(res_enrich[geneset_id, "gs_genes"], ","))
+      thisset_members_ids <- annotation_obj$gene_id[match(thisset_members, annotation_obj$gene_name)]
+    }
+  }else {
+    # overwritable via a list
+    if (!all(genelist %in% rownames(res_de))) {
+      not_there <- genelist[!(genelist %in% rownames(res_de))]
+      warning("Some of the provided gene ids were not found in the SummarizedExperiment",
+              "\nNot found: ", not_there)
+    }
+    thisset_members_ids <- intersect(rownames(res_de), genelist)
+    thisset_name <- "Custom list"
+  }
+
+  thisset_members_ids
+  
+  # Prepare the data
+  complete_genes_ids <- rownames(res_de)
+  complete_genes <- annotation_obj$gene_name[match(complete_genes_ids, annotation_obj$gene_id)]
+  
+  padj_complete <- res_de[complete_genes_ids, "padj"]
+  filter_info_complete <- sapply(padj_complete, function(x) x <= FDR)
+  padj_complete <- sapply(padj_complete, function(x) -log10(x))
+  
+  log2FoldChange_complete <- res_de[complete_genes_ids, "log2FoldChange"]
+  
+  gene_set_belong <- complete_genes_ids %in% thisset_members_ids
+  filter_info_complete <- filter_info_complete & gene_set_belong
+
+  
+  thisset_complete_data <- data.frame(complete_genes_ids, padj_complete, log2FoldChange_complete, filter_info_complete, gene_set_belong)
+  colnames(thisset_complete_data) <- c("genes", "logTransformedpvalue", "log2FoldChange", "significant", "belonging")
```

Comment:
feel free to indent a la 
```
data.frame(
  complete_genes_ids,
  padj_complete,
  ...
)
```

It is better to avoid too long lines - it is a harmless thing, but sometimes it helps to also "give you an idea" of the structure.